src/v/raft/consensus.cc

```diff
@@ -584,6 +584,14 @@ void consensus::dispatch_recovery(follower_index_metadata& idx) {
 }
 
 ss::future<result<model::offset>> consensus::linearizable_barrier() {
+    if (!is_leader()) {
+        // Callers may expect that our offsets (including commit offset) are
+        // all correct after their barrier completes, so do not allow them
+        // to barrier if we are in the state where we are elected but have
+        // not yet finished replicating the first batch of the new term.
+        co_return raft::errc::not_leader;
+    }
+
```

Comment:
This is correct, however it should happen under the lock. It is acquired few lines later

Reply:
Makes sense, updated